core/src/main/java/com/yahoo/ycsb/Client.java

```diff
@@ -835,6 +916,48 @@ public void run()
       }
     }
 
+    //run warmup threads
+    int warmupopcount = Integer.parseInt(props.getProperty(WARMUP_OPERATION_COUNT_PROPERTY, "0"));
+    int warmupexectime = Integer.parseInt(props.getProperty(WARMUP_EXECUTION_TIME, "0"));
+
+    if(dotransactions && (warmupopcount > 0 || warmupexectime > 0))
+    {
+      System.err.println("Starting warmup the datastore...");
+      CountDownLatch warmupLatch=new CountDownLatch(threadcount);
+      List<WarmupThread> warmupThreads = new ArrayList<WarmupThread>();
+      for (int threadid = 0; threadid < threadcount; threadid++)
+      {
+        DB db = null;
+        try
+        {
+          db = DBFactory.newDB(dbname, props);
+        } catch (UnknownDBException e)
+        {
+          System.out.println("Unknown DB " + dbname);
+          System.exit(0);
+        }
+        int operations_perthread = warmupopcount / threadcount;
+        if (threadid < (warmupopcount % threadcount))
+        {
+          ++operations_perthread;
+        }
+        WarmupThread t = new WarmupThread(db, dotransactions, workload, props,
+                operations_perthread, targetperthreadperms, warmupLatch, warmupexectime);
+        warmupThreads.add(t);
+      }
+      for (Thread t : warmupThreads)
+      {
+        t.start();
+      }
+      try {
+        warmupLatch.await();
+      } catch (InterruptedException e) {
+        e.printStackTrace();
```

Comment:
that sounds like the join operation is wrong. at a minimum when catching an interruption exception we need to set the thread's interrupted status.

additionally, if we've been interrupted we're supposed to "clean up and get out." that probably means the remaining work we do after is a bad idea.